src/Search/Api/Elasticsearch/ElasticQueryBuilder.php

```diff
@@ -180,6 +180,12 @@ public function searchFor(string $string) : QueryBuilder
         return $this;
     }
 
+    private function escapeReservedChars(string $string) : string
+    {
+        // Reintroduce support for some of the characters when they can be appropriately handled.
+        return preg_replace(['@[><]@', '@[-+=&|!(){}[\]^"~*?:\\\/]@'], ['', addslashes('\\$0')], $string);
```

Comment:
if some of this was copied from stackoverflow ([for example](https://stackoverflow.com/questions/33845230/escape-elasticsearch-special-characters-in-php/68389722#68389722)) you should attribute it/link to where you pulled it from.

I don't think we have to be worried about licencing issues when sourcing from SO but solutions now often become superseded later and ... it's just polite.

I found that regex by following links in this ticket: 
https://github.com/elastic/elasticsearch-php/issues/620

which also has this helpful link (couldn't find an Opensearch version):
https://www.elastic.co/guide/en/elasticsearch/reference/current/query-dsl-query-string-query.html#_reserved_characters

could I get that link pasted above that regex? It would help explain what all those things are you're removing or escaping.

And I'm not entirely sure what the result of `addslashes('\\$0')` looks like but hardcoding the result might be clearer.

Reply:
> I don't think we have to be worried about licencing issues when sourcing from SO but solutions now often become superseded later and ... it's just polite.

I agree. Will add this.